arcgis-ios-sdk-samples/Maps/Manage operational layers/MMLLayersViewController.swift

```diff
@@ -15,150 +15,147 @@
-        case .added:
-            //layers in reverse order
-            let index = dataSourceIndexForIndexPath(layers, indexPath: indexPath)
-            cell.textLabel?.text = (layers[index] as AnyObject).name
-        case .removed:
-            let layer = deletedLayers[indexPath.row]
-            cell.textLabel?.text = layer.name
-        }
+        
+        let layerForIndexPath: AGSLayer? = {
+            switch Section(rawValue: indexPath.section)! {
+            case .operational:
+                return map?.operationalLayers.object(at: indexPath.row) as? AGSLayer
+            case .removed:
+                return removedLayers[indexPath.row]
+            }
+        }()
+        cell.textLabel?.text = layerForIndexPath?.name
         return cell
     }
+
+    //MARK: - UITableViewDelegate
     
     override func tableView(_ tableView: UITableView, editingStyleForRowAt indexPath: IndexPath) -> UITableViewCell.EditingStyle {
-        switch Section(indexPath.section) {
-        case .added:
+        switch Section(rawValue: indexPath.section)! {
+        case .operational:
             return .delete
         case .removed:
             return .insert
         }
     }
     
     override func tableView(_ tableView: UITableView, canMoveRowAt indexPath: IndexPath) -> Bool {
-        return indexPath.section == Section.added.rawValue
+        return Section(rawValue: indexPath.section) == .operational
     }
     
-    //update the order of layers in the array
-    override func tableView(_ tableView: UITableView, moveRowAt sourceIndexPath: IndexPath, to destinationIndexPath: IndexPath) {
-        guard destinationIndexPath != sourceIndexPath else { return }
-        //layers in reverse order
-        let sourceIndex = dataSourceIndexForIndexPath(layers, indexPath: sourceIndexPath)
-        let destinationIndex = dataSourceIndexForIndexPath(layers, indexPath: destinationIndexPath)
+    override func tableView(_ tableView: UITableView, targetIndexPathForMoveFromRowAt sourceIndexPath: IndexPath, toProposedIndexPath proposedDestinationIndexPath: IndexPath) -> IndexPath {
         
-        let layer = layers[sourceIndex] as! AGSLayer
+        if Section(rawValue: sourceIndexPath.section) == .operational,
+            Section(rawValue: proposedDestinationIndexPath.section) == .operational {
+            // only allow reordering within the operational layers section
+            return proposedDestinationIndexPath
+        }
+        return sourceIndexPath
+    }
+    
+    override func tableView(_ tableView: UITableView, moveRowAt sourceIndexPath: IndexPath, to destinationIndexPath: IndexPath) {
+        // update the order of layers in the array
         
-        layers.removeObject(at: sourceIndex)
-        layers.insert(layer, at:destinationIndex)
+        if destinationIndexPath != sourceIndexPath {
+            map?.operationalLayers.exchangeObject(at: sourceIndexPath.row, withObjectAt: destinationIndexPath.row)
```

Comment:
This line causes the following to be printed in the debugger:

> ArcGIS Runtime Error Occurred. Set a breakpoint on C++ exceptions to see the original callstack and context for this error:  Error Domain=com.esri.arcgis.runtime.error Code=24 "Object already owned." UserInfo={NSLocalizedFailureReason=Already owned., NSLocalizedDescription=Object already owned., Additional Message=Already owned.}

Have you looked into that? It doesn't look like you're doing anything wrong, but we might need to file a runtime issue for this.

Reply:
Nice catch. I don't think my code does anything special to cause that. I think you're right about filing a runtime issue, seems like it's an error with `exchangeObject(at:, withObjectAt:)` internal to the SDK.

Reply:
Yes, please make an issue in cocoa-common-api.